src/Microsoft.AspNetCore.Razor.Evolution/Legacy/HtmlMarkupParser.cs

```diff
@@ -305,7 +310,7 @@ public override void ParseBlock()
                     else
                     {
                         Context.ErrorSink.OnError(
-                            CurrentSymbol.Start,
+                            CurrentStart,
```

Comment:
In general errors all **start** at the start of the current symbol. 

Note that in this PR I changed the meaning of current location to track the end of `CurrentSymbol`.  I did a bulk find-replace on these and then corrected the ones that should still be `CurrentLocation`. The differences are mostly down to whether or not the current symbol will be accepted or put back.